### PR TITLE
Fix secret name for openshift mode

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
Upgrading the gatekeeper causes cert-issue due to the fact that the same secret name has been created by the service-ca already 
Ref: https://issues.redhat.com/browse/ACM-10859
Signed-off-by: yiraeChristineKim <yikim@redhat.com>

### CODE REVIEW

File: controllers/gatekeeper_controller.go [hunk 1]
```diff
@@ -895,7 +896,29 @@ func openShiftDeploymentOverrides(obj *unstructured.Unstructured) error {
 
 	err = unstructured.SetNestedField(obj.Object, containers, "spec", "template", "spec", "containers")
 	if err != nil {
-		return errors.Wrapf(err, "Failed to set the OpenShift overrides")
+		return errors.Wrapf(err, "Failed to set --disable-cert-rotation for the OpenShift")
```
Comment (mprahl) [inline controllers/gatekeeper_controller.go:899]:
```suggestion
		return errors.Wrapf(err, "Failed to set --disable-cert-rotation for OpenShift")
```

File: controllers/gatekeeper_controller.go [hunk 2]
```diff
@@ -895,7 +896,29 @@ func openShiftDeploymentOverrides(obj *unstructured.Unstructured) error {
...
+	}
+
+	volumes, _, err := unstructured.NestedSlice(obj.Object, "spec", "template", "spec", "volumes")
+	if err != nil || len(volumes) == 0 {
+		return errors.Wrapf(err, "Failed to set the secret name for the OpenShift")
```
Comment (mprahl) [inline controllers/gatekeeper_controller.go:904]:
```suggestion
		return errors.Wrapf(err, "Failed to set the secret name for OpenShift")
```

File: controllers/gatekeeper_controller.go [hunk 3]
```diff
@@ -895,7 +896,29 @@ func openShiftDeploymentOverrides(obj *unstructured.Unstructured) error {
...
+	}
+
+	vol, ok := volumes[0].(map[string]interface{})
+	if !ok {
+		return errors.Wrapf(err, "Failed to parse volumes")
+	}
+
+	err = unstructured.SetNestedField(vol, OpenshiftSecretName, "secret", "secretName")
+	if err != nil {
+		return errors.Wrapf(err, "Failed to set secretName for the OpenShift")
```
Comment (mprahl) [inline controllers/gatekeeper_controller.go:914]:
```suggestion
		return errors.Wrapf(err, "Failed to set the certificate volume mount for OpenShift")
```

File: controllers/gatekeeper_controller.go [hunk 4]
```diff
@@ -895,7 +896,29 @@ func openShiftDeploymentOverrides(obj *unstructured.Unstructured) error {
...
+	}
+
+	volumes[0] = vol
+
+	err = unstructured.SetNestedField(obj.Object, volumes, "spec", "template", "spec", "volumes")
+	if err != nil {
+		return errors.Wrapf(err, "Failed to set the volumes with the openshift secretName for the OpenShift")
```
Comment (mprahl) [inline controllers/gatekeeper_controller.go:921]:
```suggestion
		return errors.Wrapf(err, "Failed to set the certificate volume mount for OpenShift")
```

**Comment by mprahl** [comment]
@yiraeChristineKim this looks good to me after the log messages are fixed.

File: test/e2e/gatekeeper_controller_test.go [hunk 5]
```diff
@@ -557,6 +557,8 @@ var _ = Describe("Gatekeeper", func() {
 	Describe("Test in Openshift Env", Label("openshift"), Ordered, Serial, func() {
 		const openshiftRoutePath = "../resources/gatekeeper_test/openshift-route-crd.yaml"
 		const openshiftNamespace = "openshift-gatekeeper-system"
+		const openshiftSecretName = "gatekeeper-webhook-server-cert-ocp"
```
Comment (dhaiducek) [inline test/e2e/gatekeeper_controller_test.go:560]: Why not use `controller.OpenshiftSecretName` here, since it's a published variable?

**Review by dhaiducek — CHANGES_REQUESTED** [review]
I think this PR should be in `main` and _then_ cherry-picked to release-3.14? (You should be able to edit the target branch.)

**Comment by yiraeChristineKim** [comment]
> I think this PR should be in `main` and _then_ cherry-picked to release-3.14? (You should be able to edit the target branch.)

Great!

**Comment by dhaiducek** [comment]
Oops--it might need a rebase. Sorry about that--I was hoping it'd be more straightforward. 😞

**Pull request closed without merging**